LibreNMS/Component.php

```diff
@@ -41,6 +41,29 @@ class Component
         'error'     => '',
     );
 
+    public function getComponentCount($device_id = null)
+    {
+        if (is_null($device_id)) {
+            // SELECT type, count(*) as count FROM component GROUP BY type
+            $SQL = "SELECT `type` as `name`, count(*) as count FROM `component` GROUP BY `type`";
+            $rows = dbFetchRows($SQL, array());
+        } else {
+            $SQL = "SELECT `type` as `name`, count(*) as count FROM `component` WHERE `device_id` = ? GROUP BY `type`";
+            $rows = dbFetchRows($SQL, array($device_id));
+        }
+
+        if (!isset($rows)) {
```

Comment:
Why not just `if (isset($rows)) {` do you 'We found some, lets re-process to make more accessible' bit then outside of the `if` block just `return false;` to cover the if not matching.